dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -447,12 +691,19 @@ def read_metadata(
         if len(parts):
             parts[0]["common_kwargs"] = {"categories": categories_dict or categories}
 
+        if len(parts) and len(parts[0]["piece"]) == 1:
+
+            # Strip all partition-dependent or unnecessary
+            # data from the `ParquetFile` object
+            pf.row_groups = None
+            pf.fmd.row_groups = None
```

Comment:
Probably only this line is needed, following the update to getstate in ParquetFile